### ISSUE TITLE
System.InvalidCastException: Specified cast is not valid when running through Wine

### ISSUE THREAD

**Issue by tjbp** [crosire/scripthookvdotnet#867]
Am attempting to run on Linux with Wine/Proton (Steam), and get the following exception:

```Unhandled Exception:
System.InvalidCastException: Specified cast is not valid.
  at GTA.ScriptDomain.LoadAssembly (System.String filename, System.Reflection.Assembly assembly) [0x00058] in <05ddb5e530f94b409069938b27151b12>:0 
  at GTA.ScriptDomain.LoadAssembly (System.String filename) [0x000ba] in <05ddb5e530f94b409069938b27151b12>:0 
  at GTA.ScriptDomain.Load (System.String path) [0x002ac] in <05ddb5e530f94b409069938b27151b12>:0 
  at ScriptHookVDotNet.Init () [0x00040] in <05ddb5e530f94b409069938b27151b12>:0 
  at <Module>.ManagedInit () [0x00185] in <7b06004b3b37431ea590b299857c72cf>:0 
  at (wrapper native-to-managed) <Module>.ManagedInit()
```

I've got 4.7.2 of the .NET framework installed - but either way System.Reflection.Assembly.loadFrom should always return an Assembly object or throw an exception, so I'm at a loss here...

Haven't successfully compiled this under Linux yet so unable to add debugging stuff - has anyone else got any ideas?

**Comment by justalemon** [comment]
I feel that this is related to #736

**Comment by tjbp** [comment]
> I feel that this is related to #736

Are you sure? I appreciate the exception is the same but I'm testing with v2.10.9 and #736 is v3 which looks like a rewrite for C#?

**Comment by crosire** [comment]
Same as #870, should have been fixed in 1f7cf37e9433a070da905846724ad80290dd39a3.
